File: math/4.py
class Solution:
        def longestPalindrome(self, s: str) -> str:
                    
                    def expand(s, l, r):
                                    while l >= 0 and r < len(s) and s[l] == s[r]:
                                                        l -= 1
                                                                        r += 1
                                                                                    return r - l - 1
                                                                                        
                                                                                        if s == None or len(s) < 1:
                                                                                                        return ""
                                                                                                            
                                                                                                            start, end = 0, 0
                                                                                                                    for i in range(len(s)):
                                                                                                                                    l1 = expand(s, i, i)
                                                                                                                                                l2 = expand(s, i, i+1)
                                                                                                                                                            maxLen = max(l1, l2)
                                                                                                                                                                        if maxLen > (end - start + 1):
                                                                                                                                                                                            start = i - (maxLen-1) // 2
                                                                                                                                                                                                            end = i + maxLen // 2
                                                                                                                                                                                                                    if end == len(s)-1:
                                                                                                                                                                                                                                    return "".join(s[start:])
                                                                                                                                                                                                                                        return "".join(s[start:end+1])
                                                                                                                                                                                                                                                

File: math/4.java
class Solution {
	    public String longestPalindrome(String s) {
		            if (s == null || s.length() < 1) {
				                return "";
						        }
			            
			            int start = 0;
				            int end = 0;
					            for (int i = 0; i < s.length(); i++) {
							                int l1 = expand(s, i, i);
									            int l2 = expand(s, i, i+1);
										                int tmp = Math.max(l1, l2);
												            if (tmp > end - start + 1) {
														                    start = i - (tmp - 1) / 2;
																                    end = i + tmp / 2;
																		                }
													            }
						            if (end == s.length() - 1)
								                return s.substring(start);
							            return s.substring(start, end + 1);
								        }
	        
	        private int expand(String s, int l, int r) {
			        while (l >= 0 && r < s.length() && s.charAt(l) == s.charAt(r)) {
					            l--;
						                r++;
								        }
				        return r - l - 1;
					    }
}
